File: src/test/java/com/raychenon/hackerrank/warmup/RepeatedStringTest.java
package com.raychenon.hackerrank.warmup;

import org.junit.BeforeClass;
import org.junit.Test;

import static junit.framework.TestCase.assertEquals;

/**
 * User: raychenon
 * Date: 13/4/19
 * https://www.hackerrank.com/challenges/repeated-string/problem
 */
public class RepeatedStringTest {

    private static RepeatedString repeatedString;

    @BeforeClass
    public static void setUp() {
        repeatedString = new RepeatedString();
    }


    @Test
    public void assertforLetterARepeatedXTimes() {
        assertEquals(RepeatedString.repeatedString("a", 10000000000000L), 10000000000000L);
    }

    @Test
    public void assertForStrinNotContainingA() {
        assertEquals(RepeatedString.repeatedString("bcdf", 10000000000000L), 0L);
    }
    
    @Test
    public void assertForStringLengthWithRemainder() {
        assertEquals(RepeatedString.repeatedString("aba", 10L), 7L);
    }


}

File: src/test/kotlin/com/raychenon/kotlin/hackerrank/warmup/RepeatedStringTest.kt
package com.raychenon.kotlin.hackerrank.warmup

import org.junit.Assert
import org.junit.Test

/**
 * User: raychenon
 * Date: 14/4/19
 * https://www.hackerrank.com/challenges/repeated-string/problem
 */
class RepeatedStringTest {

    @Test
    fun assertforLetterARepeatedXTimes() {
        Assert.assertEquals(10000000000000, RepeatedString.repeatedString("a", 10000000000000))
    }

    @Test
    fun assertForStrinNotContainingA() {
        Assert.assertEquals(0, RepeatedString.repeatedString("bcdf", 1343))
    }

    @Test
    fun assertPairingSock2() {
        Assert.assertEquals(7, RepeatedString.repeatedString("aba", 10))
    }

    @Test
    fun assertEmpty() {
        Assert.assertEquals(2, RepeatedString.repeatedString("avatar", 3))
    }
}